core/src/executor/context/row_context.rs

```diff
@@ -103,6 +118,25 @@ impl<'a> RowContext<'a> {
             Self::Bridge { left, right } => {
                 [left.get_all_entries(), right.get_all_entries()].concat()
             }
+            _ => vec![],
+        }
+    }
+}
+
+impl<'a> From<(&'a [String], &'a DataRow)> for RowContext<'a> {
+    fn from((columns, data_row): (&'a [String], &'a DataRow)) -> Self {
+        match data_row {
+            DataRow::Vec(values) => Self::RefVecData { columns, values },
+            DataRow::Map(values) => Self::RefMapData(values),
+        }
+    }
+}
```

Comment:
It appears that implementing this code using RowContext's methods to add rows would be more preferable than using From. Since Map does not require columns, it would be more desirable to split it into functions that receive different parameters.